### PR TITLE
DTLS Size Fix

### PR DESCRIPTION
**Pull request by ejohnstown**
When attempting to send a message with DTLS, if it is too large, return an error rather than splitting it across records. (ZD 10602)

### CODE REVIEW

File: wolfssl/error-ssl.h [hunk 1]
```diff
@@ -167,6 +167,7 @@ enum wolfSSL_ErrorCodes {
     CLIENT_CERT_CB_ERROR         = -436,   /* Client cert callback error */
     SSL_SHUTDOWN_ALREADY_DONE_E  = -437,   /* Shutdown called redundantly */
     TLS13_SECRET_CB_E            = -438,   /* TLS1.3 secret Cb fcn failure */
+    DTLS_SIZE_ERROR              = -439,   /* Trying to send too much data */
```
Comment (dgarske) [inline wolfssl/error-ssl.h:170]: Does this need to be added internal.c `wolfSSL_ERR_reason_error_string`?

File: src/internal.c [hunk 2]
```diff
@@ -17709,9 +17709,11 @@ int SendData(WOLFSSL* ssl, const void* data, int sz)
 
         len = wolfSSL_GetMaxRecordSize(ssl, sz - sent);
 
-#ifdef WOLFSSL_DTLS
-        if (IsDtlsNotSctpMode(ssl)) {
-            len = min(len, MAX_UDP_SIZE);
```
Comment (dgarske) [inline src/internal.c:17714]: Just confirming this check was moved to `wolfSSL_GetMaxRecordSize`?

File: src/internal.c [hunk 3]
```diff
@@ -17709,9 +17709,11 @@ int SendData(WOLFSSL* ssl, const void* data, int sz)
...
+#if defined(WOLFSSL_DTLS) && !defined(WOLFSSL_NO_DTLS_SIZE_CHECK)
```
Comment (dgarske) [inline src/internal.c:17712]: Should we add a test case for `DTLS_SIZE_ERROR` with `!WOLFSSL_NO_DTLS_SIZE_CHECK`?

File: src/internal.c (hunk 2 above)
Comment (ejohnstown) [reply src/internal.c:17714]: Yes.

File: wolfssl/error-ssl.h (hunk 1 above)
Comment (ejohnstown) [reply wolfssl/error-ssl.h:170]: It is in the function.

File: src/internal.c (hunk 3 above)
Comment (ejohnstown) [reply src/internal.c:17712]: Added a test case for this.

**Comment by dgarske** [comment]
@ejohnstown :

```
examples/client/client.c:596:25: warning: Value stored to 'ret' is never read
                        ret = err;
                        ^     ~~~
```

**Comment by ejohnstown** [comment]
I have a fix for that. The sanitize-address is giving me a lot of headache at the moment.

**Pull request merged**